Add TOC terms (#208)
grafana/writers-toolkit: docs/sources/writing-guide/documentation-structure/index.md

# Documentation structure

> The Grafana Labs Docs team makes intentional decisions about how to organize and structure product documentation. With well-structured content, users can find what they need quickly and easily.

Reviewer: ```suggestion
The Grafana Labs Docs team makes intentional decisions about how to organize and structure product documentation. With well-structured content, you can find what you need quickly and easily.
```
Shifting this to communicate directly to the reader.

Reviewer 2: Can we have a fancier name for documentatoin structure? Maybe documentation or content framework? like we have the content development model?

Reviewer: Does "fancier" depart from plain language?

Reviewer: This is guidance intended to be action-oriented and answers the question, "Where do I put stuff?" when writing documentation.

For user-journey specifics, would it be okay with you that we address that in a subsequent PR? For this PR, it feels like internal scope creep, and we can always create a follow-on PR with a user-journey focus.

Reviewer: I went ahead and created #210, so that we can cover that angle with the attention that it needs.

Author: @brendamuir The original page title is Documentation structure. We can certainly reconsider the page title. Maybe when we look at the context with user journey? I've added a note in the comments on issue #210 gathering your comments.

Before you begin contributing to documentation, it is important to understand the structure of the content.

According to STC’s [Information Design and Information Architecture: Why Technical Communicators Should Care About These Fields](https://www.stc.org/intercom/2022/05/information-design-and-information-architecture-why-technical-communicators-should-care-about-these-fields/),
information architecture is "...the practice of organizing, structuring, and labeling information to it’s easy to find, use, and understand..."

> Generally, documentation structure determines how content is:

Reviewer: Not coming up with an alternative to this incomplete phrase that stands alone. How can line 22 be made complete on its own?

Author: Line 22 is the introduction to the three bullets that follow underneath it. Those bullets incorporated into a complete sentence. I can add punctuation to make that clearer, if you like.

Reviewer: Yes, that is clear. It is a small anti-pattern wrt style that I will leave as-is for this PR.

Reviewer: Also, it has a nice zen ring to it as is. :)

- Titled
- Grouped
- Combined (or not combined) with other, related content

The following examples are based on the Grafana OSS and Enterprise documentation.

## Structure of published content

> Using the same content structure across documentation provides consistent experience. Topics are listed from high level to more specific. For example, a new Grafana user may wish to learn conceptual information first, so Introduction is listed before installation.

Reviewer: ```suggestion
A content structure that spans sets of documentation provides a consistent experience.
Information flows from a higher (less specific) level to a lower (more specific) level.
For example, a new Grafana user wants to learn conceptual information first, so _Introduction_ comes before _Installation_.
```

Reviewer: (I also introduced semantic line breaks here.)

Reviewer 2: Can we add some notes about a content framework supporting customers along their entire journey - from pre-sales thru config thru upgrade - to link this to our model?

Reviewer: See #210.

> Nat all Topics are not used for every product. For example, Grafana OSS may use all of the headings, while Grafana Enterprise Traces only uses a subset.

Reviewer: ```suggestion
You do not have to use all of the topics for every product.
For Grafana OSS for example, you might use all of the headings.
For Grafana Enterprise Traces for example, you might only use a subset of topics.
```

Reviewer: Shifted to communicating directly to the reader.

Reviewer: Shifted to communicating directly to the reader.

> This table provides a list of the high-level topics used for grouping content. For example, a conceptual page explaining metrics would go under the Introduction topic.
> 
> Italicized topics are optional and are usually found in specific contexts. For example, the Create, Manage, Monitor topics are used in Grafana OnCall but are not used in Grafana Tempo.

Reviewer: ```suggestion
The following table contains high-level topics for groups of content. For example, a conceptual page that explains metrics goes under the _Introduction_ topic.
Some topics are optional and are usually found in specific contexts. For example, the _Create_, _Manage_, _Monitor_ topics are used in Grafana OnCall, but are not used in Grafana Tempo.
```

Reviewer: I removed the baked-in italics comment because it runs contrary to style for citing titles of sections, headings, books, articles, etc.

Author: Okay. I had used the italics to differentiate topics that were specific to a subset of products, instead of being universal. It looks like you've addressed those items in the comments in the table.

Author: Here's what I did for this page: 
- Italicized topics, as you were doing, to highlight them in the text. Not using bold since they aren't UI elements. 
- Links to example pages are not italicized, because it goes against the style guides (see previous comment).

Reviewer: The phrase "Italicized topics are optional" might throw folks off if _foo_ gets formatted differently via CSS by the Design and/or Website team at some point. It could become purple and very large, and italicized. Highly unlikely example to illustrate a point. :)

> | Topic | Example link | Contains |

Reviewer: ```suggestion
| Topic | Example link | Contents |
```
Use a substantive for the heading because it stands alone grammatically.

| --- | --- | --- |

> | Introduction | Introduction to Grafana | Conceptual information, fundamentals, architecture, etc. |

Reviewer: ```suggestion
| Introduction | Introduction to Grafana | Conceptual, fundamental, or architectural information. |
```

Reviewer 2: What can the product do for me?
User journey: Pre-sales / anyone wanting to dig deeper on what the features offer
Technical and educational content that supports grafana.com

Reviewer 2: Do we need a separate issue for that? It's just adding a couple of bullet points about what an intro is

Reviewer: I think it would be helpful to have the headspace to look at the docs through that lens.

> | Get started | Get started with Tempo | Opinionated walk-throughs and tutorials |

Reviewer: ```suggestion
| Get started | Get started with Tempo | Opinionated walk-throughs and tutorials. |
```

> | Set up | Set up Loki | System requirements, Set up, configure, upgrade, migrate, etc. |

Reviewer: ```suggestion
| Set up | Set up Loki | System requirements, and subsections titled _Set up_, _Configure_, _Upgrade_, or _Migrate_. |
```

Reviewer 2: How do I implement the product?
Includes both install and configure 
User journey: Implementation and adoption

Reviewer: See #210.

| Configure |  | Configure may be it’s own directory if the number of pages |

> | Create alerts | Create alerts for Grafana OnCall | Specific to operational products such as Grafana OnCall. Not used with backend database products like Tempo and Loki. |

Reviewer: ```suggestion
| _Create alerts_ | _Create alerts for Grafana OnCall_ | Specific to operational products, such as Grafana OnCall. Do not use with backend database products, such as Tempo and Loki. Use _Alerts_ instead, and refer to an operational product for details. |
```
Am I correct in assuming that you mean that the source of truth lies elsewhere?

> | Manage alerts | Manage alerts for Grafana OnCall | Specific to operational products such as Grafana OnCall. Not used with backend database products like Tempo and Loki. |

Reviewer: ```suggestion
| _Manage alerts_ | _Manage alerts for Grafana OnCall_ | Specific to operational products, such as Grafana OnCall. Do not use with backend database products, such as Tempo and Loki. Use _Alerts_ instead, and refer to an operational product for details. |
```

Reviewer 2: How do I use the product to meet my goals?

Reviewer 2: Can we use SLOs as an example here - since we already have content to support those headings and can even link. so - Create, Manage, Monitor SLOs. And say "specific to Grafana Ops products (Alerting, Oncall, Incident, SLOs)

> | Monitor alerts | Monitor alerts for Grafana OnCall | Specific to operational products such as Grafana OnCall. Not used with backend database products like Tempo and Loki. |

Reviewer: ```suggestion
| _Monitor alerts_ | _Monitor alerts for Grafana OnCall_ | Specific to operational products, such as Grafana OnCall. Do not use with backend database products, such as Tempo and Loki. Use _Alert_ instead, and refer to an operational product for details.  |
```

> | Integrate (with) product or Send data |  | How to set up data integrations, product integrations, data sources, clients, plugins, etc. |

Reviewer: ```suggestion
| Integrate (with) [product] &#124; Send data |  | How to set up data integrations, product integrations, data sources, clients, plugins, and more. |
```
I assume the "with" is useful if you are integrating something with a Grafana Labs product. Can we simply remove it?

Reviewer 2: with is definitely a use case; please leave

> | Query data | TraceQL query editor | Query languages, query tools, examples |

Reviewer: ```suggestion
| _Query data_ | _TraceQL query editor_ | Query languages, query tools, and examples. |
```
This example might need a verb... "_Edit TraceQL queries_"

Reviewer: And (HOW) to do that, one uses the TraceQL query editor, which can be described in the first paragraph of that topic.

> | Visualize data |  | Dashboard concepts and procedures |

Reviewer: ```suggestion
| _Visualize data_ |  | Dashboard concepts and procedures. |
```

Reviewer 2: This section should be product-specific--if it's about dashboard concepts and procedures, topics should link to the dashboard docs?

Reviewer: > This section should be product-specific--if it's about dashboard concepts and procedures, topics should link to the dashboard docs?

I revised my comment to make it easier for Kim to accept it. Does that do the trick, @brendamuir?

Reviewer 2: Yep-thanks!

> | Alert |  |  |

Reviewer: ```suggestion
| _Alert_ |  |  |
```

Reviewer: Let’s tie _Alert_ to _Create alert_ to make sure folks know when to use _Alert_.

Getting through the rest of my comments and can noodle on this for a bit...

> | Monitor [product] |  | Information about using tools to monitor a Grafana product. |

Reviewer: ```suggestion
| _Monitor [product]_ |  | Information about using tools to monitor a Grafana Labs product. |
```

> | References | HTTP API Reference | API, configuration references, SDKs, etc. Material that is usually not procedural and infrequently used. |

Reviewer: ```suggestion
| _References_ | _HTTP API reference_ | APIs, configuration references, SDKs, and more. Material that is usually not procedural. |
```
I removed "infrequently" because configuration parameters are used frequently, as are APIs.

Reviewer 2: Here we should comment that you only need a folder "References" if there are multiple references; if you only have one or two APIs; simply list them.

Reviewer: This will be covered in a _Guiding principles_ topic, presumably, because that advice applies to all sections.

# Documentation structure

The Grafana Labs Docs team makes intentional decisions about how to organize and structure product documentation. With well-structured content, users can find what they need quickly and easily.

Before you begin contributing to documentation, it is important to understand the structure of the content.

According to STC’s [Information Design and Information Architecture: Why Technical Communicators Should Care About These Fields](https://www.stc.org/intercom/2022/05/information-design-and-information-architecture-why-technical-communicators-should-care-about-these-fields/),
information architecture is "...the practice of organizing, structuring, and labeling information to it’s easy to find, use, and understand..."

Generally, documentation structure determines how content is:

- Titled
- Grouped
- Combined (or not combined) with other, related content

The following examples are based on the Grafana OSS and Enterprise documentation.

## Structure of published content

> A content structure that spans sets of documentation provides a consistent experience.

Reviewer 3: ```suggestion
A standardized content structure that spans sets of documentation provides a consistent user experience.
```

Information flows from a higher (less specific) level to a lower (more specific) level.
For example, a new Grafana user wants to learn conceptual information first, so _Introduction_ comes before _Installation_.

> You do not have to use all of the topics for every product.

Reviewer 3: ```suggestion
Depending on your product design and maturity, you may not need every topic listed below.  If a topic does not apply to your project, you don't need to use it.  
```

For Grafana OSS for example, you might use all of the headings.
For Grafana Enterprise Traces for example, you might only use a subset of topics.

> This table provides a list of the high-level topics used for grouping content. For example, a conceptual page explaining metrics would go under the Introduction topic.

Reviewer 3: ```suggestion
This table provides a list of the high-level topics used for grouping content. When writing new content, consider where it should appear given this content structure. For example, a conceptual page explaining metrics would go under the Introduction topic.
```

> Italicized topics are optional and are usually found in specific contexts. For example, the Create, Manage, Monitor topics are used in Grafana OnCall but are not used in Grafana Tempo.

Author: ```suggestion
Some topics are optional and are usually found in specific contexts. For example, the Create, Manage, Monitor topics are used in Grafana OnCall but are not used in Grafana Tempo.
```

# Documentation structure

The Grafana Labs Docs team makes intentional decisions about how to organize and structure product documentation. The topic levels discussed on this page reflect common user goals. For example, a first-time Grafana user needs to understand the basic concepts before installing the product.

Before you begin contributing to documentation, it is important to understand the structure of the content.

## User goals and documentation structure

> Any documentation you create should be focused on what your user's goals are. Refer to [Focus on user goals]({{< relref "../../style-guide/style-conventions" >}}) for more information.

Reviewer: ```suggestion
Focus any documentation that you write on what your user's goals are. For more information, refer to [Focus on user goals]({{< relref "../../style-guide/style-conventions" >}}).
```

> Your documentation structure should reflect these user goals. Think about what your users want to do, what they need to know, and how they can accomplish the tasks.

Reviewer: ```suggestion
Use the structure of your documentation to reflect the user’s goals. Think about what your users want to do, what they need to know, and how they can accomplish the tasks.
```

With well-structured content, you can find what you need quickly and easily. Topics flow in a logical progression.

<!-- Commenting this paragraph out until we have an actual section on information architecture.

According to STC’s [Information Design and Information Architecture: Why Technical Communicators Should Care About These Fields](https://www.stc.org/intercom/2022/05/information-design-and-information-architecture-why-technical-communicators-should-care-about-these-fields/),
information architecture is "...the practice of organizing, structuring, and labeling information to it’s easy to find, use, and understand..."
-->

## Structure of published content

Generally, documentation structure determines how content is:

- Titled
- Grouped
- Combined (or not combined) with other, related content

A standardized content structure that spans sets of documentation provides a consistent user experience.
Information flows from a higher (less specific) level to a lower (more specific) level.
For example, a new Grafana user wants to learn conceptual information first, so _Introduction_ comes before _Installation_.

### Use the topics you need

> Depending on your product design and maturity, you may not need every topic listed below.
> If a topic does not apply to your project, you don't need to use it.
> For Grafana OSS for example, you might use all of the headings.
> For Grafana Enterprise Traces for example, you might only use a subset of topics.

Reviewer: ```suggestion
Depending on your product design and maturity, you may not need every topic:
* If a topic does not apply to your project, you don't need to use it.
* For Grafana OSS for example, you might use all of the headings.
* For Grafana Enterprise Traces for example, you might only use a subset of topics.
```

Some topics are optional and are usually found in specific contexts.

> For example, the _Create_, _Manage_, and _Monitor_ topics are used in Grafana OnCall but are not used in Grafana Tempo.

Reviewer: ```suggestion
For example, the _Create_, _Manage_, and _Monitor_ topics are used in Grafana OnCall, but are not used in Grafana Tempo.
```

### Topic list

> This table provides a list of the high-level topics used for grouping content. When writing new content, consider where it should appear given this content structure. For example, a conceptual page explaining metrics would go under the Introduction topic.

Reviewer: ```suggestion
You can use the following high-level topics to group content. When writing new content, consider where it should appear given this content structure. For example, a conceptual page explaining metrics would go under the Introduction topic.
```
Rationale: I lean towards talking less about the mechanics, such as "page", "table", "article", and more about the content.

| Topic | Example link | Contents |
| --- | --- | --- |
| _Introduction_ | [Introduction to Grafana](/docs/grafana/latest/introduction/) | Conceptual, fundamental, or architectural information.  |
| _Get started_ | [Get started with Tempo](/docs/tempo/latest/getting-started/) | Opinionated walk-throughs and tutorials. |
| _Set up_ | [Set up Grafana](/docs/grafana/latest/setup-grafana/) | System requirements, and subsections titled _Set up_, _Configure_, _Upgrade_, or _Migrate_. |
| _Configure_ | [Configure Grafana Agent](/docs/agent/latest/configuration/) | _Configure_ can be its own section directory if the number of pages warrants it. Making this determination is not an exact science; use your best judgement. |

> | _Create alerts_ | [Create a Grafana managed alert rule](/docs/grafana-cloud/alerting/alerting-rules/create-grafana-managed-rule/) | Specific to Grafana Ops products (Alerting, Oncall, Incident, SLOs). The word _alert_ may be changed, depending upon the product. If used with Grafana SLO, this then topic would be _Create SLO_. Do not use with backend database products, such as Tempo and Loki. Use _Alerts_ instead, and refer to an operational product for details. |
> | _Manage alerts_ | | Specific to Grafana Ops products (Alerting, Oncall, Incident, SLOs). Do not use with backend database products, such as Tempo and Loki. Use _Alert_ instead, and refer to an operational product for details. |
> | _Monitor alerts_ | | Specific to Grafana Ops products (Alerting, Oncall, Incident, SLOs). Do not use with backend database products, such as Tempo and Loki. Use _Alert_ instead, and refer to an operational product for details.  |

Author: ```suggestion
| _Create alerts_ | [Create a Grafana managed alert rule](/docs/grafana-cloud/alerting/alerting-rules/create-grafana-managed-rule/) | Specific to Grafana Ops products (Alerting, OnCall, Incident, SLOs). The word _alert_ may be changed, depending upon the product. If used with Grafana SLO, this then topic would be _Create SLO_. Do not use with backend database products, such as Tempo and Loki. Use _Alerts_ instead, and refer to an operational product for details. |
| _Manage alerts_ | [Manage and monitor SLOs](/docs/grafana-cloud/slo/manage/) | Specific to Grafana Ops products (Alerting, OnCall, Incident, SLOs). Do not use with backend database products, such as Tempo and Loki. Use _Alert_ instead, and refer to an operational product for details. |
| _Monitor alerts_ | [Meta monitoring](/docs/grafana/next/alerting/meta-monitoring/) | Specific to Grafana Ops products (Alerting, OnCall, Incident, SLOs). Do not use with backend database products, such as Tempo and Loki. Use _Alert_ instead, and refer to an operational product for details.  |
```
